test/test-mongocrypt-marking.c

```diff
@@ -896,7 +902,212 @@ static void test_mc_get_mincover_from_FLE2RangeFindSpec(_mongocrypt_tester_t *te
+        g_count++;
+        ASSERT(BSON_ITER_HOLDS_DOCUMENT(&g_iter));
+        uint32_t subdoc_len;
+        const uint8_t *subdoc_buf;
+        bson_t subdoc;
+        bson_iter_document(&g_iter, &subdoc_len, &subdoc_buf);
+        ASSERT(bson_init_static(&subdoc, subdoc_buf, subdoc_len));
+
+        bson_iter_t sub_iter;
+        bson_iter_init_find(&sub_iter, &subdoc, "p");
+        ASSERT(BSON_ITER_HOLDS_BINARY(&sub_iter));
+        bson_iter_binary(&sub_iter, NULL, &p_len, &p_data);
+        ASSERT(p_len == 16 + expectedPLength);
+
+        if (useRangeV2) {
+            _mongocrypt_buffer_t p_buf, decrypted_buf;
+            ASSERT(_mongocrypt_buffer_copy_from_data_and_size(&p_buf, p_data, p_len));
+            _mongocrypt_buffer_init_size(&decrypted_buf, expectedPLength);
+
+            // Decrypt p. If useRangeV2, the last byte should be 0 or 1, depending on whether isLeaf.
+            uint32_t decrypted_bytes;
+            ASSERT(fle2alg->do_decrypt(crypt->crypto,
+                                       NULL,
+                                       mc_ECOCToken_get(ecocToken),
+                                       &p_buf,
+                                       &decrypted_buf,
+                                       &decrypted_bytes,
+                                       status));
+            ASSERT(decrypted_bytes == expectedPLength);
+            if (decrypted_buf.data[decrypted_buf.len - 1] == 1) {
+                leaf_count++;
+            } else {
+                ASSERT(decrypted_buf.data[decrypted_buf.len - 1] == 0)
+            }
+
+            _mongocrypt_buffer_cleanup(&decrypted_buf);
+            _mongocrypt_buffer_cleanup(&p_buf);
+        }
+    }
+    ASSERT(g_count == expectedEdges);
+    if (useRangeV2) {
+        // There should be exactly one leaf in any insert call.
+        ASSERT(leaf_count == 1);
+    }
+    bson_destroy(&ciphertextBSON);
+    mongocrypt_status_destroy(status);
+}
+
+// Get the ECOC token to use in decryption.
+static mc_ECOCToken_t *getECOCToken(mongocrypt_t *crypt) {
+    mongocrypt_status_t *status = mongocrypt_status_new();
+    // Test token key that we added earlier is all zeros.
+    _mongocrypt_buffer_t tokenKey;
+    _mongocrypt_buffer_init_size(&tokenKey, MONGOCRYPT_TOKEN_KEY_LEN);
+    memset(tokenKey.data, 0, MONGOCRYPT_TOKEN_KEY_LEN);
+
+    mc_CollectionsLevel1Token_t *collectionsLevel1Token =
+        mc_CollectionsLevel1Token_new(crypt->crypto, &tokenKey, status);
+    mc_ECOCToken_t *ecocToken = mc_ECOCToken_new(crypt->crypto, collectionsLevel1Token, status);
+    ASSERT(mongocrypt_status_ok(status));
+
+    mc_CollectionsLevel1Token_destroy(collectionsLevel1Token);
+    _mongocrypt_buffer_cleanup(&tokenKey);
+    mongocrypt_status_destroy(status);
+    return ecocToken;
+}
+
+static void test_mc_marking_to_ciphertext(_mongocrypt_tester_t *tester) {
+    // Test that whether range V2 is enabled or disabled, the ciphertext matches our expectations.
```

Comment:
```suggestion
    if (!_aes_ctr_is_supported_by_os) {
        printf("Common Crypto with no CTR support detected. Skipping.");
        return;
    }
    
    // Test that whether range V2 is enabled or disabled, the ciphertext matches our expectations.
```

I expect this will resolve task failures on macOS 10.14. macOS 10.14 does not support CTR mode encryption (MONGOCRYPT-440).